ccl/backupccl: enable restore from metadata SST

rhu713 opened the pull request:
In 22.1, the metadata SST was introduced and written along the manifest during
every backup. This SST had separate key-values for all repeated fields in the
manifest so that they can be streamed in future backup and restore jobs to
reduce the memory used by those jobs. This patch introduces the ability to
restore from the manifest SST and avoids the need to maintain copies of
repeated fields such as files and descriptors in memory. The largest
improvement from this change compared to the previous work done with slim
manifests is via the memory saved from having to load all descriptors from all
manifests in an incremental chain.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/95845)
<!-- Reviewable:end -->

Commit:
ccl/backupccl: enable restore from metadata SST

In 22.1, the metadata SST was introduced and written along the manifest during
every backup. This SST had separate key-values for all repeated fields in the
manifest so that they can be streamed in future backup and restore jobs to
reduce the memory used by those jobs. This patch introduces the ability to
restore from the manifest SST and avoids the need to maintain copies of
repeated fields such as files and descriptors in memory. The largest
improvement from this change compared to the previous work done with slim
manifests is via the memory saved from having to load all descriptors from all
manifests in an incremental chain.

Release note: None
